src/compiler/program.ts

```diff
@@ -2346,6 +2346,10 @@ namespace ts {
                 programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Invalid_value_for_reactNamespace_0_is_not_a_valid_identifier, options.reactNamespace));
             }
 
+            if (options.jsxFactory && options.reactNamespace) {

```

Comment:
should add another check as well that `jsxFactory` can not be used without `jsx: react`